modules/nf-google/src/main/nextflow/cloud/google/pipelines/GooglePipelinesTaskHandler.groovy

```diff
@@ -109,11 +108,34 @@ class GooglePipelinesTaskHandler extends TaskHandler {
         if (!task.container) {
             throw new ProcessUnrecoverableException("No container image specified for process $task.name -- Either specify the container to use in the process definition or with 'process.container' value in your config")
         }
-        if (!instanceType) {
-            throw new ProcessUnrecoverableException("No instance type specified for process $task.name -- Please provide a 'cloud.instanceType' definition in your config")
+    }
+
+    //
+    // Use the process cpus and memory directives to create custom GCP instance
+    // If memory not specified, default to 1 GB per cpu.  An absence of the cpus directive defaults to 1 cpu.
+    // If the process machineType is defined, use that instead of cpus/memory (must be a predefined GCP machine type)
+    // The deprecated cloud.instanceType will be used if specified and machineType is not specified
+    String getProcessMachineType() {
+        String machineType = getProcessMachineType(executor.getSession().config.navigate("cloud.instanceType"), task.config.machineType, task.config.cpus, task.config.memory)
+
+        log.trace "[GPAPI] Task: $task.name - Instance Type: $machineType"
+
+        return machineType
+    }
+
+    String getProcessMachineType(String cloudInstanceType, String taskMachineType, int cpus, MemoryUnit memory) {
+
+        String machineType = taskMachineType ?: cloudInstanceType
+
+        if (machineType == null) {
+            long megabytes = memory != null ? memory.mega : cpus*1024
+            machineType = 'custom-' + cpus + '-' + megabytes
         }
```

Comment:
While it makes sense to have the memory requirement to be related to the number of cpus, nextflow does not do this for other executors. For consistency, it would be better to fallback to 1Gb irrespective the number of cpus. The user should be responsible to provide a specific value.

Reply:
This makes sense in the abstract and I can move forward with this if you like as I agree the user should specify the desired memory requirements.  

However, the GCP API by default requires at least 0.9 GB per CPU.  So, anything above 1 cpus will simply fail when the API request is issued under this scenario.

Let me know if you'd still like me to proceed with this change.

Reply:
OK, in that case, I agree that makes sense to accommodate the requirements of the platform. Please put a note in the google pipelines docs regarding this.

Reply:
Updated docs.